pkg/git/git.go

```diff
@@ -0,0 +1,80 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	giturls "github.com/whilp/git-urls"
+)
+
+func run(args ...string) error {
+	cmd := exec.Command("git", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func HasStagedChange() error {
+	err := run("diff", "--staged", "--exit-code")
+	if err != nil {
+		return errors.Errorf("repository contains staged changes: %q", err)
+	}
+	return nil
+}
+
+func RmRecursive(path string) error {
+	return run("rm", "-r", path)
+}
+
+func AddAll(path string) error {
+	return run("add", "-A", path)
+}
+
+func SubtreeAdd(path, repository, revision string) error {
+	return run("subtree", "add", "--squash", "--prefix", path, repository, revision)
+}
+
+func Commit(message string) error {
+	log.Info("Committing the changes...")
```

Comment:
Consider printing the commit message.  Might be useful for the end user when debugging.

Reply:
and the commit hash maybe? or even `git show -s`?

Reply:
it's already display something line:
```
[master fe94d3c] message
 1 file changed, 1 insertion(+), 1 deletion(-)
```
to stdout